### PR TITLE
content browser

### PR DESCRIPTION
**Pull request by Minghui-Gao**
display assets of game project
drag and drop assets into scene view

### CODE REVIEW

**Comment by Minghui-Gao** [comment]
image
no open project

**Comment by Minghui-Gao** [comment]
image
open a project

**Comment by Minghui-Gao** [comment]
image
show the project directory in the project view and assets in the content browser

**Comment by Minghui-Gao** [comment]
image
can open folder by clicking the folder in project view or content browser. all the files under the folder you clicked wiil be displayed in the content browser and if png file will display the preview

**Comment by TejasJD** [comment]
> image can open folder by clicking the folder in project view or content browser. all the files under the folder you clicked wiil be displayed in the content browser and if png file will display the preview

Hi,
Nice work.
Can you do that for all the sprites in the sprite sheet using sprite class. That'll show all the sprites from sprite sheet as separate assets.

**Comment by Minghui-Gao** [comment]
> > image can open folder by clicking the folder in project view or content browser. all the files under the folder you clicked wiil be displayed in the content browser and if png file will display the preview
> 
> Hi, Nice work. Can you do that for all the sprites in the sprite sheet using sprite class. That'll show all the sprites from sprite sheet as separate assets.

yeah.Will try to implement this by the end of the week

**Comment by Minghui-Gao** [comment]
image
can show all the sprites now. But the problem is it can only show correctly with the same size of sprite sheet. So is it possible to get the size of sprites and how many sprites will be in each row and column when creating the sprite table? @TejasJD

**Comment by TejasJD** [comment]
For now, we need to provide data for size.
What we can do though for later Is that we can have a Sprite sheet generator, that takes in X images and generates Sprite sheet of desired format.

**Comment by Minghui-Gao** [comment]
image
image
image
image
For now, When users add a sprite sheet, they need to provide the sprite format file (<name of sprite sheet>.meta). In this way, the size and coordinates of each sprite are known at the time of creating the sprite sheet.So now can show all the sprites from different size sprite sheets

**Comment by TejasJD** [comment]
Nice work, this should do the job for now. I think the one of the options we can consider is the one where, you keep checking the texture directory for changes. For example, Let the texture directory be a directory with textures of size 16 x 16 (or images of whatever size the user wishes to put).
Only if change happens in the directory (that is when an image is added or removed, we update generate a new sprite sheet with all the images in the directory).
So, although user would see all texture files separate in the folder, the program would generate a sprite sheet from all the files for internal usage.
That way users won't have to find and create a sprite sheet on their own. They can just put the images in the texture assets directory as they want without knowing about the sprite sheet.

**Review by TejasJD — APPROVED** [review]

File: HznApplication/HznApp.h [hunk 1]
```diff
@@ -9,13 +9,20 @@ class EditorLayer : public Hzn::Layer
...
+	std::string assetPath;
 	std::string projectPath ="Project (No open project)";;
 	std::vector<std::shared_ptr<Hzn::TreeNode<std::string>>> nodes;
 	std::string contextObject = "";
 	bool openContext = false;
 	bool dirOpenContext = false;
+	bool clickStatus = false;
 	std::shared_ptr<Hzn::GameObject> copiedGameObject;
 	std::map<std::string, Hzn::AudioSource*> audioFileMap;
+	std::filesystem::path m_CurrentDirectory;
```
Comment (LiamANeeson) [inline HznApplication/HznApp.h:21]: Really good work!

**Review by LiamANeeson — APPROVED** [review]
LGTM!

File: HorizonEngine/Platform/OpenGL/FDWindows.cpp [hunk 2]
```diff
@@ -94,6 +95,36 @@ namespace Hzn {
 				return std::string();
 			}
 		}
+
+		std::string FileDialogs::openFolder() {
```
Comment (alex-mulkerrins) [inline HorizonEngine/Platform/OpenGL/FDWindows.cpp:99]: cool to see different features intertwining here!

**Review by alex-mulkerrins — APPROVED** [review]

**Pull request merged**